hoomd/md/pytest/test_pair.py

```diff
@@ -0,0 +1,939 @@
+                      [-0.0922397, -0.00225115]]
+
+    param_list = []
+    for pair_potential in params.keys():
+        kT_dict = {}
+        if pair_potential == "DPD":
+            kT_dict = {"kT": 2}
+        elif pair_potential == "DPDLJ":
+            kT_dict = {"kT": 1}
+        for i in range(3):
+            param_list.append((pair_potential,
+                               getattr(hoomd.md.pair, pair_potential),
+                               kT_dict,
+                               params[pair_potential][i],
+                               forces[pair_potential][i],
+                               energies[pair_potential][i]))
+    return param_list
+
+
+@pytest.fixture(scope="function",
+                params=_forces_and_energies(),
+                ids=(lambda x: x[0]))
+def forces_and_energies(request):
+    return deepcopy(request.param)
+
+
+def _assert_equivalent_type_params(type_param1, type_param2):
+    for pair in type_param1:
+        if isinstance(type_param1[pair], dict):
+            for key in type_param1[pair]:
+                np.testing.assert_allclose(type_param1[pair][key],
+                                           type_param2[pair][key])
+        else:
+            assert type_param1[pair] == type_param2[pair]
+
+
+def _assert_equivalent_parameter_dicts(param_dict1, param_dict2):
+    for key in param_dict1:
+        assert param_dict1[key] == param_dict2[key]
+
+
+def test_rcut(simulation_factory, two_particle_snapshot_factory):
+    lj = hoomd.md.pair.LJ(nlist=hoomd.md.nlist.Cell(), r_cut=2.5)
+    lj.params[('A', 'A')] = {'sigma': 1, 'epsilon': 0.5}
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj.r_cut[('A', 'A')] = 'str'
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj.r_cut[('A', 'A')] = [1, 2, 3]
+
+    sim = simulation_factory(two_particle_snapshot_factory(dimensions=3, d=.5))
+    integrator = hoomd.md.Integrator(dt=0.005)
+    integrator.forces.append(lj)
+    integrator.methods.append(hoomd.md.methods.Langevin(hoomd.filter.All(),
+                                                        kT=1, seed=1))
+    sim.operations.integrator = integrator
+
+    _assert_equivalent_type_params(lj.r_cut.to_dict(), {('A', 'A'): 2.5})
+    sim.operations.schedule()
+    sim.run(1)
+    _assert_equivalent_type_params(lj.r_cut.to_dict(), {('A', 'A'): 2.5})
+
+
+@pytest.mark.parametrize("mode", ['none', 'shifted', 'xplor'])
+def test_mode(simulation_factory, two_particle_snapshot_factory, mode):
+    cell = hoomd.md.nlist.Cell()
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj = hoomd.md.pair.LJ(nlist=cell, r_cut=2.5, mode=1)
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj = hoomd.md.pair.LJ(nlist=cell, r_cut=2.5, mode='str')
+    with pytest.raises(hoomd.typeconverter.TypeConversionError):
+        lj = hoomd.md.pair.LJ(nlist=cell, r_cut=2.5, mode=[1, 2, 3])
```

Comment:
This is getting tested for every parameterized value. It may be worth having two tests for this.